daemon/emer-daemon.c

```diff
@@ -1123,6 +1145,169 @@ set_max_bytes_buffered (EmerDaemon *self,
   priv->max_bytes_buffered = max_bytes_buffered;
 }
 
+static void schedule_next_midnight_tick (EmerDaemon *self);
+static void schedule_next_month_tick (EmerDaemon *self);
+
+static void
+save_aggregate_timers_to_tally (EmerDaemon *self,
+                                const char *date)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+  EmerAggregateTimerImpl *timer_impl;
+  GHashTableIter iter;
+  gint64 now_monotonic_us;
+
+  now_monotonic_us = g_get_monotonic_time ();
+
+  g_hash_table_iter_init (&iter, priv->aggregate_timers);
+  while (g_hash_table_iter_next (&iter, (gpointer *) &timer_impl, NULL))
+    {
+      emer_aggregate_timer_impl_store (timer_impl,
+                                       date,
+                                       now_monotonic_us);
+      emer_aggregate_timer_impl_split (timer_impl,
+                                       now_monotonic_us);
+    }
+}
+
+static EmerTallyIterResult
+buffer_daily_aggregate_event_to_queue (guint32     unix_user_id,
+                                       GVariant   *event_id,
+                                       GVariant   *aggregate_key,
+                                       GVariant   *payload,
+                                       guint32     counter,
+                                       const char *date,
+                                       gpointer    user_data)
+{
+  EmerDaemon *self = user_data;
+
+  emer_daemon_record_aggregate_event (self,
+                                      event_id,
+                                      date,
+                                      counter,
+                                      payload != NULL,
+                                      payload);
+
+  return EMER_TALLY_ITER_CONTINUE;
+}
```

Comment:
We had discussed not doing this step (which transfers data from the on-disk tally into the possibly in-memory submission queue) and instead reworking the code that reads from the existing submission queue to also pull from these on-disk tallies. Either one seems workable but this way is less durable.

If doing it this way,  wouldn't we want to DELETE here, once the event is in the submission queue?